### PR TITLE
deps: Add helm to dependencies

### PR DESCRIPTION
**Pull request by Callisto13**
- Use this local helm in the Tiltfile
- Also tidy up the dependencies script to make it easier to read and
  silence some shellcheck violations

### CODE REVIEW

File: tools/functions.sh [hunk 1]
```diff
@@ -137,9 +139,9 @@ validate_file() {
     local checksums_path="${1}"
     local file_path="${2}"
     local cmd="${3}"
-    if ! grep $(openssl dgst -sha256 ${file_path} | cut -d ' ' -f 2) ${checksums_path}; then
-        echo ${cmd} is not a valid file
-        rm -rf ${file_path}
+    if ! grep "$(openssl dgst -sha256 "${file_path}" | cut -d ' ' -f 2)" "${checksums_path}"; then
```
Comment (yitsushi) [inline tools/functions.sh:142]: I'm not sure what is the intended behavior here, but the I think either the `"${file_path}"` should be `\"${file_path}\"` or `'${file_path}'`.

Or can we just move out the command into its own line? Like:
```bash
local digest=$(openssl dgst -sha256 "${file_path}" | cut -d ' ' -f 2)
if ! grep "${digest}" "${checksums_path}"; then
....
```

**Review by yitsushi — CHANGES_REQUESTED** [review]
Looks good, the only thing I used the "request changes", that grep with quotes looks sus

File: tools/download-deps.sh [hunk 2]
```diff
@@ -70,30 +71,37 @@ download_dependency() {
         "${bin_dir}"/stoml "${dependencies_toml}" "${tool}"."${property}"
     }
 
-    local url_and_path
-    local fetch
-    local binarypath
+    local tarpath
+    tarpath=$(instantiate_url "$(run_stoml tarpath)")
```
[the remark this answers is not in the document]
Comment (yitsushi) [reply tools/download-deps.sh:75]: Feel free to ignore this.

File: tools/functions.sh (hunk 1 above)
Comment (Callisto13) [reply tools/functions.sh:142]: yeh i think i prefer that idea

File: tools/download-deps.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (Callisto13) [reply tools/download-deps.sh:75]: eh, just grouping things for readability, i found it hard to follow before, it's quite a large func

**Review by luizbafilho — APPROVED** [review]
